app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -423,6 +437,37 @@ void OpenXRInputSource::UpdateHaptics(ControllerDelegate &delegate)
     CHECK_XRCMD(applyHapticFeedback(mHapticAction, duration, XR_FREQUENCY_UNSPECIFIED, pulseIntensity));
 }
 
+bool OpenXRInputSource::GetHandTrackingInfo(const XrFrameState& frameState, XrSpace localSpace) {
+    if (OpenXRExtensions::sXrLocateHandJointsEXT == XR_NULL_HANDLE || mHandTracker == XR_NULL_HANDLE)
+        return false;
+
+    // @FIXME: We currently require XR_FB_hand_tracking_aim to show beam and pointer target
+    if (!OpenXRExtensions::IsExtensionSupported(XR_FB_HAND_TRACKING_AIM_EXTENSION_NAME))
+        return false;
+
+    hasHandJoints = false;
+
+    // Update hand locations
+    XrHandJointsLocateInfoEXT locateInfo { XR_TYPE_HAND_JOINTS_LOCATE_INFO_EXT };
+    locateInfo.baseSpace = localSpace;
+    locateInfo.time = frameState.predictedDisplayTime;
+
+    XrHandJointLocationsEXT jointLocations { XR_TYPE_HAND_JOINT_LOCATIONS_EXT };
+    jointLocations.jointCount = XR_HAND_JOINT_COUNT_EXT;
+    jointLocations.jointLocations = mHandJoints.data();
+    jointLocations.next = XR_NULL_HANDLE;
+
+    mAimState.status = 0;
+    mAimState.next = jointLocations.next;
+    jointLocations.next = &mAimState;
```

Comment:
The logic here is a bit confusing. I think it's better to do
```
     mAimState.status = 0;
     mAimState.next = XR_NULL_HANDLE;
```

and then
```
     XrHandJointLocationsEXT jointLocations { XR_TYPE_HAND_JOINT_LOCATIONS_EXT };
     jointLocations.jointCount = XR_HAND_JOINT_COUNT_EXT;
     jointLocations.jointLocations = mHandJoints.data();
     jointLocations.next = &mAimState;
```

Reply:
The idea here was to make it more future-proof. If in the future we link another structure in `jointLocations.next` before this code, we don't break it here, nor need to remember to touch this code when that happens.

Not a big deal, will change it if you prefer.

Reply:
done